src/dataframe/dataframe.jl

```diff
@@ -1223,34 +1236,59 @@ Base.convert(::Type{DataFrame}, A::AbstractMatrix) = DataFrame(A)
 
 Base.convert(::Type{DataFrame}, d::AbstractDict) = DataFrame(d, copycols=false)
 
-function Base.push!(df::DataFrame, row::Union{AbstractDict, NamedTuple}; columns::Symbol=:equal)
-    if !(columns in (:equal, :intersect))
-        throw(ArgumentError("`columns` keyword argument must be `:equal` or `:intersect`"))
+function Base.push!(df::DataFrame, row::Union{AbstractDict, NamedTuple}; cols::Symbol=:setequal,
+                    columns::Union{Nothing,Symbol}=nothing)
+    if !isnothing(columns)
```

Comment:
This is better for inference IIRC:
```suggestion
    if columns !== nothing
```